packages/models/src/store/merge-strategy.ts

```diff
@@ -0,0 +1,266 @@
+import { get, mergeWith, set } from 'lodash/fp'
+import { NoCellToUseException, NonExistentException } from '../exceptions'
+import { deepForIn } from '../utils'
+import { OutPointString } from './interface'
+
+export abstract class MergeStrategy<S, T = S> {
+  abstract merge(object: T, source?: S): T
+  abstract findAndUpdate(updateInfo: {
+    paths?: string[]
+    value: unknown
+    outPointStrings: OutPointString[]
+    states: Record<OutPointString, S>
+    isValueEqual(value: unknown, compare: unknown): boolean
+    isSimpleType(value: unknown): boolean
+  }): {
+    update?: {
+      outPointString: OutPointString
+      state: S
+    }[]
+    remove?: OutPointString[]
+  }
+}
+
+export class UseLatestStrategy<S> extends MergeStrategy<S, S> {
+  merge(object: S, _source?: S): S {
+    return object
+  }
+
+  findAndUpdate(updateInfo: {
+    paths?: string[]
+    value: unknown
+    outPointStrings: OutPointString[]
+    states: Record<OutPointString, S>
+  }): {
+    update?: {
+      outPointString: OutPointString
+      state: S
+    }[]
+    remove?: OutPointString[]
+  } {
+    const lastOutPoint = updateInfo.outPointStrings.at(-1)
+    if (!lastOutPoint) throw new NoCellToUseException()
+    if (!updateInfo.paths?.length) {
+      return {
+        update: [
+          {
+            outPointString: lastOutPoint,
+            state: updateInfo.value as S,
+          },
+        ],
+      }
+    }
+    const updateValueInstates = get([lastOutPoint, ...updateInfo.paths.slice(0, updateInfo.paths.length - 1)])(
+      updateInfo.states,
+    )
+    if (updateValueInstates === undefined) {
+      throw new NonExistentException(`${lastOutPoint}${updateInfo.paths.join('.')}`)
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    updateValueInstates[updateInfo.paths.at(-1)!] = updateInfo.value
+    return {
+      update: [
+        {
+          outPointString: lastOutPoint,
+          state: updateInfo.states[lastOutPoint],
+        },
+      ],
+    }
+  }
+}
+
+export class DefaultMergeStrategy<S> extends MergeStrategy<S, S> {
+  private defaultMergeWith = mergeWith((objValue, srcValue) => {
+    if (Array.isArray(objValue)) {
+      return objValue.concat(srcValue)
+    }
+  })
+
+  merge(object: S, source?: S): S {
+    if (source === undefined || source === null) return object
+    if (Array.isArray(object)) return object.concat(source) as S
+    return this.defaultMergeWith(source, object)
```

Comment:
The order of `destination` and `source` is reversed from that of definition

Reply:
We should use the new value in `object` to cover the old value in `source`, so we need set `object` as the second parameter.

Reply:
If we can switch the parameters of `private defaultMergeWith` so they won't be mixed in all cases.